src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileService.scala

```diff
@@ -92,10 +92,13 @@ class FileService(fileRepository: FileRepository,
     val filters = fileFilters.getOrElse(FileFilters())
     for {
       fileAndMetadataList <- fileRepository.getFiles(consignmentId, filters)
-      ffidMetadataList <- ffidMetadataService.getFFIDMetadata(consignmentId)
-      avList <- avMetadataService.getAntivirusMetadata(consignmentId)
-      ffidStatus <- fileStatusService.getFileStatus(consignmentId)
-    } yield fileAndMetadataList.toFiles(avList, ffidMetadataList, ffidStatus).toList
+      getFfidMetadataList: Future[List[FFIDMetadata]] = ffidMetadataService.getFFIDMetadata(consignmentId)
+      getAvList: Future[List[AntivirusMetadata]] = avMetadataService.getAntivirusMetadata(consignmentId)
+      getFfidStatus: Future[Map[UUID, String]] = fileStatusService.getFileStatus(consignmentId)
```

Comment:
Can these be put outside of the for loop? similarly to how you've done it for the other files. Just for consistency.

Reply:
Hi Thanh, I wanted to but the issue with this is that `avList`, `ffidMetadataList`, `ffidStatus` are passed into the `tofiles` method of `fileAndMetadataList`; I could be wrong but I think that we'd need to ensure that `fileAndMetadataList` is completed beforehand.